spot-contracts/contracts/RolloverVault.sol

```diff
@@ -861,34 +861,42 @@ contract RolloverVault is
 
     /// @dev Executes a system-level rebalance. This operation transfers value between
     ///      the perp reserve and the vault such that the system moves toward balance.
-    ///      Settles protocol fees, charged as a percentage of the bi-directional flow in value.
     ///      Performs some book-keeping to keep track of the vault and perp's assets.
     function _rebalance(IPerpetualTranche perp_, IERC20Upgradeable underlying_) private {
-        // If the vault has any perp tokens we deconstruct it
-        // such that it's value reflects in the vault's TVL,
-        // before the rebalance calculation.
+        // Claim mint/burn fees collected by perp.
+        perp_.claimFees(address(this));
         _meldPerps(perp_);
 
         SubscriptionParams memory s = _querySubscriptionState(perp_);
-        RebalanceData memory r = feePolicy.computeRebalanceData(s);
-        if (r.underlyingAmtIntoPerp <= 0) {
-            // We transfer value from perp to the vault, by minting the vault perp tokens (without any deposits).
-            // NOTE: We first mint the vault perp tokens, and then pay the protocol fee.
-            perp_.rebalanceToVault(r.underlyingAmtIntoPerp.abs() + r.protocolFeeUnderlyingAmt);
-
-            // We immediately deconstruct perp tokens minted to the vault.
-            _meldPerps(perp_);
-        } else {
-            // We transfer value from the vault to perp, by minting the perp tokens (after making required deposit)
-            // and then simply burning the newly minted perp tokens.
-            uint256 perpAmtToTransfer = r.underlyingAmtIntoPerp.toUint256().mulDiv(perp_.totalSupply(), s.perpTVL);
-            _trancheAndMintPerps(perp_, underlying_, s.perpTVL, s.seniorTR, perpAmtToTransfer);
-            IERC20Burnable(address(perp_)).burn(perpAmtToTransfer);
+        int256 underlyingAmtIntoPerp = feePolicy.computeRebalanceAmount(s);
+
+        // We rebalance from perp to the vault.
+        perp_.rebalanceToVault(underlyingAmtIntoPerp);
```

Comment:
Should we put a conditional around this? I think it's safe to say that if amt into perp >= 0, we don't need to tell perp to send vault anything. It would also save us the meld operation.

Reply:
👍🏽